pgjdbc/src/main/java/org/postgresql/core/NativeQuery.java

```diff
@@ -48,24 +54,36 @@ public NativeQuery(String nativeSql, int[] bindPositions, boolean multiStatement
    * @return a human-readable representation of this query
    */
   public String toString(@Nullable ParameterList parameters) {
-    if (bindPositions.length == 0) {
+    if (!parameterCtx.hasParameters()) {
       return nativeSql;
     }
 
     int queryLength = nativeSql.length();
-    String[] params = new String[bindPositions.length];
-    for (int i = 1; i <= bindPositions.length; ++i) {
-      String param = parameters == null ? "?" : parameters.toString(i, true);
+    String[] params = new String[parameterCtx.nativeParameterCount()];
+    for (int i = 1; i <= parameterCtx.nativeParameterCount(); ++i) {
+      final String param;
+      if (parameters != null) {
+        param = parameters.toString(i, true);
+      } else {
+        if (parameterCtx.hasNamedParameters()) {
+          param = ":" + parameterCtx.getPlaceholderName(i);
```

Comment:
Stringbuilder here ?

Reply:
I'm not sure it is worth it? It is would require a StringBuilder instance to do a single append.

Reply:
Thinking about this, + is an implicit StringBuilder. I think String.concat would yield the same per of new objects created. But my guess would be this is negligible as it is only incurred when a new SQL text is parsed.

Reply:
@kimjand , check `bindName` and `appendBindName`